### DISCUSSION TITLE
Is it possible adding brief text for each unit for each day?

### DISCUSSION THREAD

**Discussion by federicopeyrani1969** [orbat-mapper/orbat-mapper#146] [Ideas]
Hi Magne!
I am beginning with the Sicilian Campaign in a local storage.
I have a pdf (attached down here) with the history for 45th US ID. 
It's a brief diary for all the three Infantry Regiments of the Division (157, 179 and 180 US IR) and subunits with positions and actions. 
I think it would be very handy to paste the text somewhere and let it appear in a ballon, so that I can follow what happens for each regiment each day. It's useful when creating the unit movement. Do you think it's possible now, or it has to be implemented?
I thought I could do something like that adding "Scenario events", but it's not working, I can't add any event. Maybe that's not been implemented, too.

Here the pdf
[US 45 ID Part 1.pdf](https://github.com/orbat-mapper/orbat-mapper/files/11265793/US.45.ID.Part.1.pdf)

I see the power of your project! It's very cool!
Regards 
Federico

**Comment by kjellmf** [comment]
Thank you for the suggestion. This is not possible right now, but definitely something I should add. It is supported by the internal model, but not exposed in the user interface yet. The only thing you can do at the moment is to change the title of an event (see screenshot below). Currently the only way to add a unit event/state is to change the scenario date and move the unit on the map. I have to find a better user interface for this.  
 
![image](https://user-images.githubusercontent.com/29217/233142656-0191c0db-5f5b-4418-a6b3-3f7b56273274.png)

**Reply by kjellmf** [reply]
Thank you for sharing the PDF. Really cool resource.

**Comment by federicopeyrani1969** [comment]
Hi Magne,
I suggest, if possible, to add the possibility of inserting linear-text unit histories. I mean, not only a brief text for each position on the map, but a whole linear text, a source of info to be then manually "parsed" into a "cloud" of points in the space/time of the orbat-mapper's map. This would be very useful for me as most of my sources are linear text. It's a parallel thing to the shorter text asked up here, I think they would both help my work.
Regards
Federico

**Reply by kjellmf** [reply]
Good idea! For this to work the text needs to be semi structured with tags or something similar to mark dates and locations. ORBAT mapper already supports Markdown so if you could use Markdown with some extra markup this could work.

**Reply by federicopeyrani1969** [reply]
In my guess there are three principal things to mark in a linear text: time, space, and actors. 
Do you know a way to mark them in Markdown?
I could try one fast example if you can suggest me how to mark those three princiapl categories.

**Reply by kjellmf** [reply]
Markdown does not have an official syntax for this, so we probably have to create a custom syntax extension.

Here are some ideas:

```markdown
# [157TH RCT](@unit)

## July 10, 1943

Landed on beaches [Green 2](@place) and [Yellow 2](@place), against
light opposition; advanced inland. [1st Battalion](@unit)
captured [S CRUCE, CAMERINA](@place) at [1445)(@time). [2nd Battalion
](@unit) captured [DUNNA FUGATA](@place) at [1715](@time), 2nd and 3rd moved on
COMISO to positions [shown on Map No 1](@location 30N 34.5E).
```

Not pretty, but something like this could work.
